modules/mobile/app/models/mobile/v0/adapters/lighthouse_individual_claims.rb

```diff
@@ -0,0 +1,187 @@
+              current_phase_back: attributes.dig('claimPhaseDates', 'currentPhaseBack'),
+              requested_decision: attributes['evidenceWaiverSubmitted5103'],
+              claim_type: attributes['claimType'],
+              contention_list: attributes['contentionList'],
+              va_representative: nil,
+              events_timeline:,
+              updated_at: updated_at(events_timeline, phase_change_date)
+            }
+          )
+        end
+        # rubocop:enable Metrics/MethodLength
+
+        private
+
+        def phase(latest_phase)
+          PHASE_TYPE_TO_NUMBER[latest_phase.to_sym]
+        end
+
+        def updated_at(events_timeline, phase_change_date)
+          dates = events_timeline.pluck(:date)
+          dates << Date.strptime(phase_change_date, '%Y-%m-%d')
+          dates.compact.max
+        end
+
+        def events_timeline(attributes)
+          events = [
+            create_event_from_string_date(:filed, attributes['claimDate']),
+            create_event_from_string_date(:completed, attributes['closeDate'])
+          ]
+
+          # Do the 8 phases
+          (1..8).each do |n|
+            events << create_event_from_string_date(
+              "phase#{n}", attributes.dig('claimPhaseDates', 'previousPhases', "phase#{n}CompleteDate")
+            )
+          end
+
+          # Add tracked items
+          events += create_events_for_tracked_items(attributes)
+
+          # Add documents not associated with a tracked item
+          events += create_events_for_documents(attributes)
+
+          # Make reverse chron with nil date items at the end
+          events.compact.sort_by { |h| h[:date] || Date.new }.reverse
+        end
+
+        def create_event_from_string_date(type, date)
+          return nil unless date
+
+          {
+            type:,
+            date: Date.strptime(date, '%Y-%m-%d')
+          }
+        end
+
+        def create_events_for_tracked_items(attributes)
+          attributes['trackedItems'].map do |tracked_item|
+            tracked_item_documents = attributes['supportingDocuments'].select do |document|
+              document['trackedItemId'] == tracked_item['id']
+            end
+            create_tracked_item_event(tracked_item, tracked_item_documents)
+          end
+        end
+
+        def create_events_for_documents(attributes)
+          untracked_documents = attributes['supportingDocuments'].select { |document| document['trackedItemId'].nil? }
+          documents_hash = create_documents(untracked_documents)
+          documents_hash.map do |document|
+            document.merge(type: :other_documents_list, date: document[:upload_date])
+          end
+        end
+
+        # Order of EVENT_DATE_FIELDS determines which date trumps in timeline sorting.
+        EVENT_DATE_FIELDS = %i[
+          closed_date
+          received_date
+          upload_date
+          opened_date
+          requested_date
+          suspense_date
+        ].freeze
```

Comment:
open date and requested date are using the same field but ill keep both here in case that changes in the future